pkg/scaffold/role.go

```diff
@@ -179,6 +179,7 @@ rules:
   - apps
   resources:
   - deployments
+  - deployments/finalizers
```

Comment:
I think maybe we should add a note to the user here, that if they change the way they deploy the operator they need to add a `<resource-kind>/finalizers ` here?

Reply:
> I think maybe we should add a note to the user here, that if they change the way they deploy the operator they need to add a `<resource-kind>/finalizers ` here?

Agree that we should add a comment about that.

I wonder if we can also scope down this rule to be better about least privilege. If I understand correctly, we know we'll be setting the owner reference on the operator deployment and we know the name of that deployment, so we could put this in a separate rule:
```yaml
- apiGroups:
  - apps
  resources:
  - deployments/finalizers
  resourceNames:
  - my-operator
  verbs:
  # Not sure if "update" is right, but we probably don't need "*"
  - update  
```

Reply:
Thanks for the suggestion!

> Agree that we should add a comment about that.

WDYT of adding this to the docs instead, tried to come up with a good sentence to add a comment, but couldn't come up with anything simple that makes sense as one sentence. :)

Reply:
@joelanford Will open PR for adding documentation for the above as soon as the metrics docs PR is merged, can you PTALA, thanks!